parachain/pallets/omni-account/src/lib.rs

```diff
@@ -194,14 +215,44 @@ pub mod pallet {
 			auth_type: Option<OmniAccountAuthType>,
 			result: DispatchResult,
 		},
-		/// Intent is requested
-		IntentRequested { who: T::AccountId, intent: Intent },
-		/// Intent is executed
-		IntentExecuted { who: T::AccountId, intent: Intent, result: IntentExecutionResult },
 		/// Member permission set
 		AccountPermissionsSet { who: T::AccountId, member_account_hash: H256 },
 		/// An auth token is requested
 		AuthTokenRequested { who: T::AccountId, expires_at: i64 },
+		/// Intent is accepted - we record the Intent detail (once)
+		IntentAccepted { who: T::AccountId, intent_id: IntentId, intent: Intent },
+		/// Intent is in-process
+		IntentInProcessUpdated {
+			who: T::AccountId,
+			intent_id: IntentId,
+			detail: IntentInProcessDetail,
+		},
+		/// Intent is completed
+		IntentCompleted { who: T::AccountId, intent_id: IntentId, detail: IntentCompletedDetail },
+	}
+
+	#[derive(Clone, Debug, PartialEq, Encode, Decode, TypeInfo)]
+	pub enum IntentInProcessDetail {
+		Swap(SwapInProcessDetail),
+	}
+
+	#[derive(Clone, Debug, PartialEq, Encode, Decode, TypeInfo)]
+	pub enum IntentCompletedDetail {
+		Swap(SwapCompletedDetail),
+	}
+
+	#[derive(Clone, Debug, PartialEq, Encode, Decode, TypeInfo)]
+	pub enum SwapInProcessDetail {
+		SourceChainBalanceDeducted { asset: ChainAsset, amount: u64 },
+		DestChainBalanceAdded { asset: ChainAsset, amount: u64 },
+		SingleChainSwapSubmitted { tx_hash: Vec<u8> },
+	}
+
+	#[derive(Clone, Debug, PartialEq, Encode, Decode, TypeInfo)]
+	pub enum SwapCompletedDetail {
+		// TODO
```

Comment:
What else we need here ?

Reply:
Like if there's any returned result if successful (e.g. actually received amount)

Reply:
For now I'll just use generic `Success/Failure` without extra info